mysql-test/t/percona_slow_extended_log_error.test

```diff
@@ -33,6 +33,18 @@ INSERT INTO t2(f1) VALUES (31.400191);
 --let grep_pattern = ^.*Last_errno: 0 .*\$
 --source include/log_grep.inc
 
+--echo
+--echo # PS-6774 - Unexpected error in Last_errno in case internal DD tables are
+--echo             accessed. Statement des not fail in case DD table names
+--echo             are submitted as strings in WHERE clause.
+--echo
+--let log_file=percona.slow_extended.log_error_4
+--source include/log_start.inc
+SELECT SCHEMA_NAME as Test FROM information_schema.schemata WHERE SCHEMA_NAME NOT IN ('mysql', 'performance_schema', 'information_schema');
```

Comment:
@oleksandr-kachan why does accessing DD tables names cause the err_no field to be filled? Can you please find out who fills the thd->m_da->mysql_errno?

Reply:
@satya-bodapati, to make sure we are on the same page here. The thd->get_stmt_da()->mysql_errno() is not populated in this case. The issue was with thd->last_errno before fix from https://github.com/percona/percona-server/commit/b79645143cd773cdc61c439f269c69520a2d3e31.

Reply:
Right @oleksandr-kachan , so this question should have been more relevant with your other fix. WIthout the other fix, who has set the errno?

Reply:
@satya-bodapati, it was set at https://github.com/percona/percona-server/blob/8.0/sql/sql_parse.cc#L6141

Reply:
I see, does the query fail? If it doesn't, do we reset the diagnostic area before the completion of the query?

Reply:
@satya-bodapati, query doesn't fail. As I can see, error in diagnostic area is not set in this case.

Reply:
got it oleksandar. my_error() sets it may be cleared at later point of time